### PR TITLE
Closes #888 filter on elements

### LINKED ISSUE #888: API for elements volume
Create an API to get volume for all elements on a page

### PR DESCRIPTION
**Pull request by timgl**
## Changes

- Allows another property filter type `element`, so you can do things like `{"key": "selector", "value": "div > a", "type": "element"}` anywhere there property filters.

## Checklist
- [ ] All querysets/queries filter by Team (if applicable)
- [ ] Backend tests (if applicable)
- [ ] Cypress E2E tests (if applicable)

### CODE REVIEW

**Review by EDsCODE — COMMENTED** [review]
Couldn't seem to find the new prop filter to use. I performed migration and installed the new dependency

**Comment by timgl** [comment]
@EDsCODE I originally wasn't planning on including this in this PR but why not! That's ready for QA.

Also will close #419 which is great.

File: posthog/api/paths.py [hunk 1]
```diff
@@ -85,7 +85,7 @@ def _apply_start_point(self, start_comparator: str, query_string: str, start_poi
     def _add_elements(self, query_string: str) -> str:
         element = 'SELECT \'<\'|| e."tag_name" || \'> \'  || e."text" as tag_name_source, e."text" as text_source FROM "posthog_element" e JOIN \
                     ( SELECT group_id, MIN("posthog_element"."order") as minOrder FROM "posthog_element" GROUP BY group_id) e2 ON e.order = e2.minOrder AND e.group_id = e2.group_id where e.group_id = v2.group_id'
-        element_group = 'SELECT g."id" as group_id FROM "posthog_elementgroup" g where v1."elements_hash" = g."hash"'
+        element_group = 'SELECT g."id" as group_id FROM "posthog_elementgroup" g where v1."elements_hash" = g."hash_id"'
```
Comment (EDsCODE) [inline posthog/api/paths.py:88]: There's two more places where `hash` needs to be replaced. On line 61 in the select and the join condition in the back

**Review by EDsCODE — CHANGES_REQUESTED** [review]
Just a few more edits in paths

File: posthog/api/paths.py (hunk 1 above)
Comment (timgl) [reply posthog/api/paths.py:88]: Done, also added a test.

**Comment by timgl** [comment]
@EDsCODE ready for re-review. I ended up removing the join fields, as it turns out the time it took to save actions went from 5 seconds to minutes...

File: posthog/api/element.py [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+        if key not in self.serializer_class.Meta.fields:
+            return response.Response([])
+
+        if request.GET.get('value'):
+            where = ' AND "posthog_element"."{}" LIKE %s'.format(key)
+            params.append('%{}%'.format(request.GET['value']))
+
+        # This samples a bunch of elements with that property, and then orders them by most popular in that sample
+        # This is much quicker than trying to do this over the entire table
+        values = Element.objects.raw("""
```
Comment (EDsCODE) [inline posthog/api/element.py:37]: There maybe be some reusability here with /path/elements endpoint in paths. Made ticket for followup here #954 954

**Review by EDsCODE — APPROVED** [review]

**Pull request merged**